client/src/components/Action.jsx

```diff
@@ -0,0 +1,21 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+const propTypes = {
+  action: PropTypes.func,
+  text: PropTypes.string.isRequired,
+};
+
+const defaultProps = {
+  action: null,
+  text: '',
+};
+
+const Action = ({ action, text }) => {
+  return <div onClick={action}>{text}</div>;
```

Comment:
We should generally never have a `div` with an `onClick` for accessibility reasons. If we cannot replace this with a `Touchable` for example from Threads, then at least lets add

```suggestion
  return <div role='button' onClick={action}>{text}</div>;
```

Also, I am a bit confused about why this component exists?

If you were to use this component, your code would look like this:

```jsx
<Action onClick={...} text='foo' />
```

If this component did not exist at all, you would just do this, which is actually less code and does not require an extra component file, an additional import, and an extra level of abstraction.

```jsx
<div onClick={...}>foo</div>
```

Reply:
okay, i changed this to a button and removed the Action component